cli/src/semgrep/config_resolver.py

```diff
@@ -250,6 +255,131 @@ def _load_config_from_local_path(self) -> List[ConfigFile]:
     def is_registry_url(self) -> bool:
         return self._origin == ConfigType.REGISTRY
 
+    def _project_metadata_for_standalone_scan(self) -> out.ProjectMetadata:
+        repo_name = os.environ.get("SEMGREP_REPO_NAME") or "unknown"
+        if repo_name is None:
+            raise SemgrepError(
+                f"Need to set env var SEMGREP_REPO_NAME to use `--config {self._config_path}`"
+            )
+
+        return out.ProjectMetadata(
+            semgrep_version=out.Version(__VERSION__),
+            scan_environment="semgrep-scan",
+            repository=repo_name,
+            repo_url=None,
+            branch=None,
+            commit=None,
+            commit_title=None,
+            commit_author_email=None,
+            commit_author_name=None,
+            commit_author_username=None,
+            commit_author_image_url=None,
+            ci_job_url=None,
+            on="unknown",
+            pull_request_author_username=None,
+            pull_request_author_image_url=None,
+            pull_request_id=None,
+            pull_request_title=None,
+            is_full_scan=True,  # always true for standalone scan
+        )
+
+    def _fetch_semgrep_cloud_platform_scan_config(self) -> ConfigFile:
+        """
+        Download a configuration from semgrep.dev using new /api/cli/scans endpoint
+        """
+        state = get_state()
+
+        products = [
+            out.Product.from_json(PRODUCT_NAMES[p])
+            for p in self._config_path.split(",")
+        ]
```

Comment:
```suggestion
        ]
        require_repo_name = set(products) & {SAST(), Secrets()} 
```

Reply:
Done in [fc41789](https://github.com/semgrep/semgrep/pull/9205/commits/fc41789b107c847fdf58199b621316a77509f6bc)